### ISSUE TITLE
When accessing a link on the Amazon website that requires a 302 redirect, it returns a 400 error.

### ISSUE THREAD

**Issue by chicagocong** [lexiforest/curl_cffi#430]
**Describe the bug**
When accessing a link on the Amazon website that requires a 302 redirect, it returns a 400 error.
**To Reproduce**
```
https://www.amazon.com.br/sspa/click?ie=UTF8&spc=MToxNTE2MDY5ODk3MjUzNTYzOjE3MzEwNDgzOTY6c3BfYnRmOjMwMDQwOTY0MTkyNDIwMjo6MDo6&url=%2FRel%25C3%25B3gio-Seiko-FD462W-comuta%25C3%25A7%25C3%25A3o-anal%25C3%25B3gica%2Fdp%2FB00BH83M86%2Fref%3Dsr_1_58_sspa%3F__mk_pt_BR%3D%25C3%2585M%25C3%2585%25C5%25BD%25C3%2595%25C3%2591%26dib%3DeyJ2IjoiMSJ9.kiEe7C09pDW7u4_0s0Ws3bOu7x97Fl9ruJJU44ZD6pQHik0mB9N80sjGu87zh9wbz3PhJnl6E-RyMfjj4XKsKULwa3BBCiwj_QNKqGO5dCVpC67vX2iE06vyQFAjcLSm_G43rRYUfKb-tA5py5zdiMGDwP8y6pDuwk03Q31GM3LCa2wE2p1qAsSX-LeaMpz3sDq8tFBWjlUJxM_jSVpacFrtB23CgRNZJuigNqMZebB8ar4dIyqwBJh1cxb-kd9Oom64bkAF8NEPIxiXrJm-9kw93Vdr6P_SKGFsw_xFWrE.01cHJVOD9CIlk3VlQmY6OYFyN-2W3AUnKCuCWwOF13E%26dib_tag%3Dse%26keywords%3Dseiko%26qid%3D1731048396%26sr%3D8-58-spons%26ufe%3Dapp_do%253Aamzn1.fos.25548f35-0de7-44b3-b28e-0f56f3f96147%26sp_csd%3Dd2lkZ2V0TmFtZT1zcF9idGY%26psc%3D1
```

**Versions**
 - OS: macOS Sequoia 15.0.1
 - curl_cffi version 0.7.3

**Comment by lexiforest** [comment]
It‘s probably because the URL was mistakenly decoded, which has not been fixed yet. Downgrade to 0.7.1 and see if it works.

**Comment by chicagocong** [comment]
@lexiforest 
Thank you very much for your messges. It is indeed a decoding issue. We are using version 0.7.3, which hasn’t been fixed.

**Comment by chicagocong** [comment]
> It‘s probably because the URL was mistakenly decoded, which has not been fixed yet. Downgrade to 0.7.1 and see if it works.

Thank you very much for your messges. It is indeed a decoding issue. We are using version 0.7.3, which hasn’t been fixed.

**Comment by lexiforest** [comment]
Fixed in 0.7.4.
